_posts/work/2018-01-30-hdf-in-the-cloud.md

```diff
@@ -64,40 +70,40 @@ So organizations like NASA are dumping large amounts of HDF onto Amazon's S3,
 that no one can actually read, except by downloading the entire file down to their hard drive,
 and then pulling out the particular bits that they need.
 This is inefficient.
-It misses out on the potential that cloud-hosted public data can offer our to society.
+It misses out on the potential that cloud-hosted public data can offer to our society.
 
 The rest of this post discusses a few of the options to solve this problem,
 including their advantages and disadvantages.
 
 1.  **Cloud Optimized GeoTIFF:** We can use modern and efficient formats from other domains, like Cloud Optimized GeoTIFF
 
-    **Good**: Fast, well established
+    *Good:* Fast, well established
 
-    **Bad**: Not actually sophisticated enough to handle some scientific use cases
+    *Bad:* Not actually sophisticated enough to handle some scientific use cases
 
 2.  **HDF + FUSE:** Continue using HDF, but mount cloud object stores as a file system with [Filesystem in Userspace, aka FUSE](https://en.wikipedia.org/wiki/Filesystem_in_Userspace)
 
-    **Good:** Works with existing files, no changes to the HDF library necessary
+    *Good:* Works with existing files, no changes to the HDF library necessary
 
-    **Bad:** It's complex, probably not as fast as possible, and has historically been brittle
+    *Bad:* It's complex, probably not as fast as possible, and has historically been brittle
 
 2.  **HDF + Custom Reader:** Continue using HDF, but teach it how to read from S3, GCS, ADL, ...
 
-    **Good:** Works with existing files, no complex FUSE tricks
+    *Good:* Works with existing files, no complex FUSE tricks
 
-    **Bad:** Requires changes to the HDF library and all downstream libraries (like Python wrappers).  Probably not performance optimal
+    *Bad:* Requires changes to the HDF library and all downstream libraries (like Python wrappers).  Probably not performance optimal
 
 3.  **HDF + Web Service:** Continue using HDF, but buffer it with a distributed service
 
-    **Good:** Works with existing files, probably decently fast, probably also works for efficient writes
+    *Good:* Probably decently fast, probably also works for efficient writes
 
-    **Bad:** Complex to write and to deploy.  Probably not free.  It hides our data behind someone who owns and runs a service, introducing an intermediary.
+    *Bad:* Complex to write and to deploy.  Probably not free.  It hides our data behind someone who owns and runs a service, introducing an intermediary.
 
```

Comment:
This could be broken into two sub-categories:
a) Web services that provide a web-accessible API to existing HDF5 files.  E.g. Pydap server, OPeNDAP hyrax.
b) Web services that utilize cloud-native storage.  E.g. HSDS

a) has the same problems as accessing HDF5 files directly - the files need to be copied from S3 before being opened by the server.
b) is optimal for directly accessing data in S3 but requires what's basically a 2nd copy of the data

Both have the problem the the capacity of the service needs to be sufficient to meet demand from clients.  This is esp. challenging with parallel computing libraries like Dask, since the Dask workers may send out a sudden flurry of requests to the service and overwhelm its ability to respond (or at best response time will slow down).